logproof/src/linear_relation.rs

```diff
@@ -859,6 +975,62 @@ impl LogProof {
         (alpha, beta, gamma, phi, psi)
     }
 
+    fn make_shared_commitment(
+        s_1_shared: &BitSlice,
+        rho: &Scalar,
+        h: &[RistrettoPoint],
+        u: &RistrettoPoint,
+    ) -> RistrettoPoint {
+        let mut commitment = RistrettoPoint::identity();
+
+        let s_1_shared = s_1_shared.iter().map(|b| b == true).collect::<Vec<bool>>();
+
+        commitment += s_1_shared
+            .par_iter()
+            .zip_eq(h.par_iter())
+            .filter_map(|(bit, p_i)| if *bit { Some(p_i) } else { None })
```

Comment:
Is this worth parallelizing? I don't know what the scheduling overhead is.

Reply:
It's what was there before (when the commitment was generated at once) but it is not precisely a long operation. So I don't think parallelization is buying us a lot.

Reply:
Are you making a comment because the shared part is likely to be much smaller than the non-shared part?